walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -26,38 +26,73 @@
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
     Set<String> attendees =  new HashSet<String>();
+    Set<String> optionalAttendees =  new HashSet<String>();
     int meetingDur = (int) request.getDuration();
     attendees.addAll(request.getAttendees());
+    optionalAttendees.addAll(request.getOptionalAttendees());
     if (meetingDur > TimeRange.WHOLE_DAY.duration()){
       return possibleTimes;
     }
-    if (events.isEmpty() || attendees.isEmpty()){
+    if (events.isEmpty()){
       possibleTimes.add(TimeRange.WHOLE_DAY);
       return possibleTimes;
     }
-    ArrayList<TimeRange> busyWithOverlap = getBusyTimes(events, attendees);
-    possibleTimes = findFreeTimes(busyWithOverlap, request.getDuration());
+    possibleTimes = getPossibleTimes(events, attendees, optionalAttendees, request.getDuration());
     return possibleTimes;
```

Comment:
nit. 
you can just `return getPossibleTimes(...)`

Reply:
done